runtime/src/serde_snapshot.rs

```diff
@@ -387,3 +494,117 @@ where
+
+    // Remap the deserialized AppendVec paths to point to correct local paths
+    let mut storage = fss_storage.into_iter().chain(iss_storage.into_iter())
+        .map(|(slot, mut slot_storage)| {
+            let now = Instant::now();
+            if now.duration_since(last_log_update).as_secs() >= 10 {
+                info!("{} slots remaining...", remaining_slots_to_process);
+                last_log_update = now;
+            }
+            remaining_slots_to_process -= 1;
+
+            let mut new_slot_storage = HashMap::new();
+            for storage_entry in slot_storage.drain(..) {
+                let file_name = AppendVec::file_name(slot, storage_entry.id());
+
+                let append_vec_path = unpacked_append_vec_map.get(&file_name).ok_or_else(|| {
+                    debug!("bprumo DEBUG: reconstruct_accountsdb_from_fields_incremental(): failed to get append vec path! file_name: {:?}", file_name);
+                    io::Error::new(
+                        io::ErrorKind::NotFound,
+                        format!("{} not found in unpacked append vecs", file_name),
+                    )
+                })?;
+
+                let (accounts, num_accounts) =
+                    AppendVec::new_from_file(append_vec_path, storage_entry.current_len())?;
+                let u_storage_entry = AccountStorageEntry::new_existing(
+                    slot,
+                    storage_entry.id(),
+                    accounts,
+                    num_accounts,
+                );
+                debug!("bprumo DEBUG: reconstruct_accountsdb_from_fields_incremental(): slot: {}, storage entry id: {}, append vec filename: {:?}, append vec path: {:?}", slot, storage_entry.id(), file_name, append_vec_path);
+
+                new_slot_storage.insert(storage_entry.id(), Arc::new(u_storage_entry));
+            }
+            Ok((slot, new_slot_storage))
+        })
+        .collect::<Result<HashMap<Slot, _>, Error>>()?;
+
+    // discard any slots with no storage entries
+    // this can happen if a non-root slot was serialized
+    // but non-root stores should not be included in the snapshot
+    storage.retain(|_slot, stores| !stores.is_empty());
+
+    accounts_db
+        .bank_hashes
+        .write()
+        .unwrap()
+        .insert(iss_slot, iss_bank_hash_info);
+
+    // Process deserialized data, set necessary fields in self
+    let max_id: usize = *storage
+        .values()
+        .flat_map(HashMap::keys)
+        .max()
+        .expect("At least one storage entry must exist from deserializing stream");
+
+    {
+        accounts_db.storage.0.extend(
+            storage.into_iter().map(|(slot, slot_storage_entry)| {
+                (slot, Arc::new(RwLock::new(slot_storage_entry)))
+            }),
+        );
+    }
+
+    if max_id > AppendVecId::MAX / 2 {
+        panic!("Storage id {} larger than allowed max", max_id);
+    }
+
+    accounts_db.next_id.store(max_id + 1, Ordering::Relaxed);
```

Comment:
heh, actually not sure what we use the `next_id` here in the accounts code for 😝 . But one thing that seems possible if we are getting the incremental snapshot and full snapshot from different nodes is that the `next_id` is not unique between storage entries in the incremental and full snapshot, i.e. they could potentially overlap if the two validators generating those snapshots started from snapshots with different `next_ids`. 

We might need to re-id storage entries with the same id.

Reply:
Oh interesting point! I'll look into how to do that.

Reply:
I was thinking maybe a node could get incremental snapshots from the same node it gets the full snapshot from. Would that resolve this?